### DISCUSSION TITLE
Virtual keyboard in WPF app run with touch screen (not have physical keyboard)

### DISCUSSION THREAD

**Discussion by phuocminh6464** [cefsharp/CefSharp#4478] [General]
Hi @amaitland,

I have a problem with input text because my WPF app run in Kiosk which has only touch screen and doesn't have physical keyboard. When users touch a button in WPF, a web page will open and allow users to input data. Because it is an external website and I only open it without changing anything. So how I do to input data if not have physical keyboard? Or do Cefsharp support virtual keyboard for WPF?

I'm using:
- CefSharp.Wpf.NETCore version 109.1.110
- Net 7
- x64
- Win10

Best regards,

**Comment by amaitland** [comment]
Please first respond to #4444 and #4435

**Reply by phuocminh6464** [reply]
so sorry and I have already responded them.

**Reply by amaitland** [reply]
Thank you.

The [VirtualKeyboardRequested](https://cefsharp.github.io/api/111.2.x/html/E_CefSharp_Wpf_ChromiumWebBrowser_VirtualKeyboardRequested.htm) can be used to show/hide an `OSK`.

There is a basic example within the project source. https://github.com/cefsharp/CefSharp/blob/cefsharp/112/CefSharp.Wpf.Example/TouchKeyboardWin10MainWindow.xaml.cs

**Reply by phuocminh6464** [reply]
Thank you so much, I can do it.
